### PR TITLE
Changes for submission v3.1 for qti backend (#791)

### PR DESCRIPTION
**Pull request by mohitmundhragithub**
* MLPerf submission for QTI backend for Mobile-V3.1

* Fix for invalid results for SD 4 Gen 2

* remove internal docker option

* Updating to the github paths

* Adding formatting changes

* Update QTI SDK for submission v3.1 (#793)

* Update QTI SDK for v3.1

* Fix Markdown linter issue

* Remove hard-code path in generate-apk-filename.sh

* Fixing a dead link

* Updating code smell fixes

* Adding md5sums for all the models

* Fixing security vulnerability

* Updating files

---------

### CODE REVIEW

**Comment by mohitmundhragithub** [comment]
@anhappdev, possible to check the reason for failure in pixel? I don't seem to have access.

File: flutter/flutter.mk [hunk 1]
```diff
@@ -65,9 +65,9 @@ flutter_folder_args=${flutter_data_folder_arg} ${flutter_cache_folder_arg}
 
 FIREBASE_ENV_FILE?=flutter/lib/firebase/firebase_options.env
 -include ${FIREBASE_ENV_FILE}
-export
```
Comment (anhappdev) [inline flutter/flutter.mk:68]: This does not work on macOS.

File: flutter/android/android.mk [hunk 2]
```diff
@@ -33,7 +33,7 @@ else
 endif
 
 .PHONY: flutter/android/libs/build
-flutter/android/libs/build:
+flutter/android/libs/build: flutter/libqtibackend/patchlibs
```
Comment (anhappdev) [inline flutter/android/android.mk:36]: Should `patchelf` be used in the master branch?

**Comment by anhappdev** [comment]
> @anhappdev, possible to check the reason for failure in pixel? I don't seem to have access.

It looks like some issue with the test service itself. I just ran the test again and it's passed.

File: flutter/lib/app_constants.dart [hunk 3]
```diff
@@ -110,5 +110,5 @@ class BackendId {
 }
 
 class BenchmarkSettings {
-  static const double maxDuration = 10 * 60; // 10 minutes
+  static const double maxDuration = 30 * 60; // 30 minutes
```
Comment (anhappdev) [inline flutter/lib/app_constants.dart:113]: There will be a PR to make this configurable, but for this PR, please keep this unchanged at 10 minutes.

File: flutter/android/android.mk (hunk 2 above)
Comment (mohitmundhragithub) [reply flutter/android/android.mk:36]: I think we can merge it for now, and with android ndk 25, we can remove it.

**Review by anhappdev — APPROVED** [review]

**Comment by mohitmundhragithub** [comment]
@anhappdev not sure why the iostests is failing. Looks like its expecting some value 3-15 but got 20 or something. 
Can you please have a look? This PR doesn't have any change for that.

**Comment by anhappdev** [comment]
> @anhappdev not sure why the iostests is failing. Looks like its expecting some value 3-15 but got 20 or something. Can you please have a look? This PR doesn't have any change for that.

It's the throughput test for Apple backend. It expects the throughput to be in a range. Let rerun the test sometimes and see how it goes. We can update the range if needed in:
https://github.com/mlcommons/mobile_app_open/blob/master/flutter/integration_test/expected_throughput.dart

**Pull request merged**